assets/javascripts/discourse/components/show-original-content.gjs

```diff
@@ -15,38 +19,42 @@ export default class ShowOriginalContent extends Component {
 
   @service router;
 
-  @tracked isTranslated = true;
+  @tracked showingOriginal = false;
 
   constructor() {
     super(...arguments);
-    this.isTranslated = !new URLSearchParams(window.location.search).has(
-      "show"
-    );
+    this.showingOriginal = cookie(SHOW_ORIGINAL_COOKIE);
   }
 
   @action
   async showOriginal() {
-    const params = new URLSearchParams(window.location.search);
-    if (this.isTranslated) {
-      params.append("show", "original");
+    if (this.showingOriginal) {
+      removeCookie(SHOW_ORIGINAL_COOKIE, { path: "/" });
     } else {
-      params.delete("show");
+      cookie(SHOW_ORIGINAL_COOKIE, true, {
+        path: "/",
+        expires: SHOW_ORIGINAL_COOKIE_EXPIRY,
+      });
     }
-    window.location.search = params.toString();
+
+    window.location.reload();
```

Comment:
Random thought... would calling `.refresh()` on the Ember router service be enough here? In theory that should re-run all `model()` hooks on routes? Might be a smoother experience than a full page reload?